test/e2e_source_map_test.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {assert, expect} from 'chai';
+import * as path from 'path';
+import {SourceMapConsumer} from 'source-map';
+import * as ts from 'typescript';
+
+import {Settings, toClosureJS} from '../src/main';
+import {toArray} from '../src/util';
+
+describe('source maps', () => {
+  it('composes source maps with tsc', function() {
+    const diagnostics: ts.Diagnostic[] = [];
+
+    const sources = new Map<string, string>();
+    sources.set(ts.sys.resolvePath('input.ts'), `
+      class X { field: number; }
+      let x : string = 'a string';
+      let y : string = 'another string';
+      let z : string = x + y;`);
+
+    // Run tsickle+TSC to convert inputs to Closure JS files.
+    const closure = toClosureJS(
+        {sourceMap: true} as ts.CompilerOptions, ['input.ts'], {isUntyped: false} as Settings,
+        diagnostics, sources);
+
+    if (!closure) {
```

Comment:
extract the repeated bits (asserting !closure and !compiledJS) into a helper method.